networkx/algorithms/tree/coding.py

```diff
@@ -384,6 +389,12 @@ def from_prufer_sequence(sequence):
     not_orphaned = set()
     index = u = next(k for k in range(n) if degree[k] == 1)
     for v in sequence:
+        # check the validity of the prufer sequence
+        if v < 0 or v > n - 1:
+            msg = """Invalid Prufer sequence: you may check that every element (sequence[i]) in the sequence
+                     does not exceed the length of the sequence plus 1 or ( 0 <= sequence[i] <= length(sequence)+1 )."""
+            raise nx.NetworkXError(msg)
```

Comment:
This message is a bit wordy (and would render very wide in the terminal). Perhaps it could be shortened to something like:

```python
NetworkXError(f"Values must be between 0 and len(sequence)-1, got {v}")
```